core/app/[locale]/(default)/product/[slug]/_actions/wishlist-action.ts

```diff
@@ -0,0 +1,304 @@
+'use server';
+
+import { BigCommerceAPIError, BigCommerceGQLError } from '@bigcommerce/catalyst-client';
+import { SubmissionResult } from '@conform-to/react';
+import { parseWithZod } from '@conform-to/zod';
+import { revalidateTag } from 'next/cache';
+import { getLocale, getTranslations } from 'next-intl/server';
+import { cache } from 'react';
+import { z } from 'zod';
+
+import { CreateWishlistMutation } from '~/app/[locale]/(default)/account/wishlists/_actions/mutation';
+import { newWishlist } from '~/app/[locale]/(default)/account/wishlists/_actions/new-wishlist';
+import { getSessionCustomerAccessToken } from '~/auth';
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { revalidate } from '~/client/revalidate-target';
+import { TAGS } from '~/client/tags';
+import { redirect } from '~/i18n/routing';
+import { serverToast } from '~/lib/server-toast';
+
+const VariantIdFromSkuQuery = graphql(`
+  query VariantIdFromSkuQuery($productId: Int!, $sku: String!) {
+    site {
+      product(entityId: $productId) {
+        variants(skus: [$sku]) {
+          edges {
+            node {
+              entityId
+            }
+          }
+        }
+      }
+    }
+  }
+`);
```

Comment:
Should we fetch variantId from product page query instead?

Reply:
@jorgemoya I considered it, but then we would be querying this additional data even in cases where it will go unused. The current implementation will only make the query when we need it (when adding a product to the wishlist), so it's slightly less load on our infrastructure 👍

Reply:
Awesome, makes sense